### PR TITLE
remove SeccompProfilePath since it is deprecated

### LINKED ISSUE #117045: Remove SeccompProfilePath
Look for "// TODO: Deprecated, remove after we switch to Seccomp field". Fix it. See https://github.com/kubernetes/kubernetes/pull/115898 as example.

/cc @saschagrunert 

/help
/good-first-issue
/sig node

add @saschagrunert as a reviewer when sending a PR.

(issue is a follow up from https://github.com/kubernetes/kubernetes/pull/117020/files#r1154025906)

### PR DESCRIPTION
**Pull request by kaminn**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
remove SeccompProfilePath since it is deprecated
#### Which issue(s) this PR fixes:

Fixes #117045

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by kaminn** [comment]
@bart0sh Here is a similar pr #117050 that was accepted, should I close this pr?

**Comment by bart0sh** [comment]
@kaminn no, you shouldn't close this PR. It does a little bit more than https://github.com/kubernetes/kubernetes/pull/117050, so let's keep both. If https://github.com/kubernetes/kubernetes/pull/117050 is merged before this you'll have to rebase.

File: pkg/kubelet/kuberuntime/helpers_linux_test.go [hunk 1]
```diff
@@ -560,12 +269,12 @@ func TestGetSeccompProfile(t *testing.T) {
 		{
 			description:   "pod seccomp profile set to SeccompProfileTypeLocalhost with empty LocalhostProfile returns error",
 			podSc:         &v1.PodSecurityContext{SeccompProfile: &v1.SeccompProfile{Type: v1.SeccompProfileTypeLocalhost}},
-			expectedError: "localhostProfile must be set if seccompProfile type is Localhost.",
+			expectedError: "localhostProfile must be set if seccompProfile type is Localhost",
 		},
```
Comment (bart0sh) [inline pkg/kubelet/kuberuntime/helpers_linux_test.go:273]: This and similar changes look unrelated to the PR.

File: pkg/kubelet/kuberuntime/security_context.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ func (m *kubeGenericRuntimeManager) determineEffectiveSecurityContext(pod *v1.Po
 			ReadonlyPaths: securitycontext.ConvertToRuntimeReadonlyPaths(effectiveSc.ProcMount),
 		}
 	}
+
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply pkg/kubelet/kuberuntime/security_context.go:37]: Yes please revert.

**Comment by saschagrunert** [comment]
This has been done already, the code got removed.
